lib/src/models/channel/types/announcement_thread.dart

```diff
@@ -1,17 +1,12 @@
-import 'package:nyxx/src/builders/message/message.dart';
 import 'package:nyxx/src/builders/permission_overwrite.dart';
-import 'package:nyxx/src/http/managers/message_manager.dart';
 import 'package:nyxx/src/models/channel/channel.dart';
+import 'package:nyxx/src/models/channel/text_channel.dart';
 import 'package:nyxx/src/models/channel/thread.dart';
-import 'package:nyxx/src/models/message/message.dart';
 import 'package:nyxx/src/models/permission_overwrite.dart';
 import 'package:nyxx/src/models/snowflake.dart';
 import 'package:nyxx/src/models/webhook.dart';
 
-class AnnouncementThread extends Channel implements Thread {
-  @override
-  MessageManager get messages => MessageManager(manager.client.options.messageCacheConfig, manager.client, channelId: id);
-
+class AnnouncementThread extends TextChannel implements Thread {
```

Comment:
What do you think abount changing name to `AnnouncementThreadChannel` -- since its channel

Reply:
The suffix `Thread` already implies it's a channel, and this terminology (`Thread` vs `Channel`) is used both for other channel types e.g `PublicThread` and `PrivateThread` and also in Discord's documentation.